ai21/models/ai21_base_model.py

```diff
@@ -0,0 +1,45 @@
+import re
+from typing import Any, Dict
+
+from pydantic import BaseModel, ConfigDict
+
+from ai21.models.pydantic_compatibility import IS_PYDANTIC_V2
+
+if IS_PYDANTIC_V2:
+    from pydantic.alias_generators import to_camel
+
+
+def to_camel_case(snake: str) -> str:
+    if re.match("^[a-z]+[A-Za-z0-9]*$", snake) and not re.search(r"\d[a-z]", snake):
+        return snake
+
+    pascal = re.sub("([0-9A-Za-z])_(?=[0-9A-Z])", lambda m: m.group(1), snake.title())
+    return re.sub("(^_*[A-Z])", lambda m: m.group(1).lower(), pascal)
+
+
+class AI21BaseModel(BaseModel):
+    if IS_PYDANTIC_V2:
+        model_config = ConfigDict(
+            alias_generator=to_camel,
+            populate_by_name=True,
+            protected_namespaces=(),
+        )
+    else:
+
+        class Config:
+            alias_generator = to_camel_case
+            allow_population_by_field_name = True
+
+    def to_dict(self) -> Dict[str, Any]:
+        return super().dict(by_alias=True)
+
+    def to_json(self) -> str:
+        return super().json(by_alias=True)
+
+    @classmethod
+    def from_dict(cls, obj: Any) -> "AI21BaseModel":
```

Comment:
`from_dict` parameter should be a dictionary no? why is it `Any`?

Reply:
In `model_validate` (v2) and `parse_obj` (v1) they use type `Any`, so I used it as well